### PR TITLE
Fixed #32548 -- Fixed crash when combining Q() objects with boolean expressions.

### PR DESCRIPTION
**Pull request by jonathan-golorry**
Fixes [ticket-32548](https://code.djangoproject.com/ticket/32548)

### CODE REVIEW

File: django/db/models/query_utils.py [hunk 1]
```diff
@@ -84,14 +84,10 @@ def deconstruct(self):
...
         if path.startswith('django.db.models.query_utils'):
             path = path.replace('django.db.models.query_utils', 'django.db.models')
-        args, kwargs = (), {}
-        if len(self.children) == 1 and not isinstance(self.children[0], Q):
-            child = self.children[0]
-            kwargs = {child[0]: child[1]}
-        else:
-            args = tuple(self.children)
-            if self.connector != self.default:
-                kwargs = {'_connector': self.connector}
```
Comment (felixxm) [inline django/db/models/query_utils.py:94]: See [comment](https://code.djangoproject.com/ticket/32548#comment:2).

File: tests/queries/test_q.py [hunk 2]
```diff
@@ -88,6 +85,12 @@ def test_deconstruct_nested(self):
         self.assertEqual(args, (Q(price__gt=F('discounted_price')),))
         self.assertEqual(kwargs, {})
 
+    def test_deconstruct_not_subscriptable(self):
+        q = Q(False)
```
Comment (felixxm) [inline tests/queries/test_q.py:89]: This is not a supported usage. Please add a test with conditional expressions, e.g. `Exists()`.

**Review by felixxm — COMMENTED** [review]
@jonathan-golorry Thanks for this patch :+1:

We should add an extra test for conditional expressions nested in `Q()`, e.g in `tests.expressions.tests.BasicExpressionsTests`.
```python
    def test_boolean_expression_in_Q(self):
        is_poc = Company.objects.filter(point_of_contact=OuterRef('pk'))
        self.gmbh.point_of_contact = self.max
        self.gmbh.save()
        self.assertCountEqual(Employee.objects.filter(Q(Exists(is_poc))), [self.max])
```

File: tests/queries/test_q.py (hunk 2 above)
Comment (jonathan-golorry) [reply tests/queries/test_q.py]: Done

File: django/db/models/query_utils.py [hunk 3]
```diff
@@ -84,14 +84,10 @@ def deconstruct(self):
...
+        args = tuple(self.children)
+        kwargs = {}
+        if self.connector != self.default:
+            kwargs = {'_connector': self.connector}
```
Comment (Ian-Foote) [inline django/db/models/query_utils.py:90]:
```suggestion
            kwargs['_connector'] = self.connector
```

**Comment by felixxm** [comment]
@jonathan-golorry Thanks :+1: Welcome aboard :boat: 

I added extra tests.

**Comment by felixxm** [comment]
> I added extra tests.

Ahh I cannot push to this branch :disappointed: Can you allow edits from maintainers?

**Comment by felixxm** [comment]
Superseded by #14140.

**Comment by jonathan-golorry** [comment]
@felixxm Sorry, I'm not completely familiar with github's access control. I'm not seeing the option. I think it's because I forked into my organization, not my personal account. Looks like an open issue with github https://github.com/isaacs/github/issues/1681 (It's funny that the entire issue history is just being mentioned by unrelated repositories).

**Pull request closed without merging**